source/detail/parser/x86_64.cpp

```diff
@@ -128,6 +129,29 @@ namespace smeagle::x86_64 {
     return result.str();
   }
 
+  // Get directionality from argument type
+  std::string getDirectionalityFromType(st::Type *paramType) {
+    auto dataType = paramType->getDataClass();
+    std::string dataTypeStr = st::dataClass2Str(dataType);
+
+    // If it's a pointer, we need to know what it's pointing to!
+    if (dataTypeStr == "pointer" || dataTypeStr == "reference") {
+      auto pointerType = paramType->getPointerType();
+
+      // if typo is pointer but type is primitive: imported
+      std::regex is_primitive("(int|char|bool|float|double|null|none|undefined)");
+      if (std::regex_search(pointerType->getName(), is_primitive)) {
+        return "import";
+      }
+
+      // but passed by pointer or reference and not primitive, value is unknown
+      return "unknown";
+    }
+
+    // Do we default to export?
+    return "export";
```

Comment:
I believe from here we want the default to be 'import' since we're handling symbols which are function definitions.

Reply:
okay, updated.